includes/wc-template-functions.php

```diff
@@ -557,72 +559,72 @@ function wc_get_product_class( $class = '', $product_id = null ) {
+	if ( ! is_array( $class ) ) {
+		$classes = preg_split( '#\s+#', $class );
 	} else {
-		// Ensure that we always coerce class to being an array.
-		$class = array();
+		$classes = $class;
 	}
 
-	if ( ! $post || ! $product ) {
-		return $classes;
-	}
-
-	$classes[] = 'post-' . $post->ID;
-	if ( ! is_admin() ) {
-		$classes[] = $post->post_type;
+	if ( ! $product ) {
+		return array_map( 'esc_attr', $classes );
 	}
-	$classes[] = 'type-' . $post->post_type;
-	$classes[] = 'status-' . $post->post_status;
 
-	// Post format.
-	if ( post_type_supports( $post->post_type, 'post-formats' ) ) {
-		$post_format = get_post_format( $post->ID );
+	$classes = array_merge(
+		$classes,
+		array(
+			'product',
+			'type-product',
+			'post-' . $product->get_id(),
+			'status-' . $product->get_status(),
+			wc_get_loop_class(),
+			$product->get_stock_status(),
+		),
+		wc_get_product_taxonomy_class( $product->get_category_ids(), 'product_cat' ),
+		wc_get_product_taxonomy_class( $product->get_tag_ids(), 'product_tag' )
+	);
 
-		if ( $post_format && ! is_wp_error( $post_format ) ) {
-			$classes[] = 'format-' . sanitize_html_class( $post_format );
-		} else {
-			$classes[] = 'format-standard';
-		}
+	if ( $product->get_image_id() ) {
+		$classes[] = 'has-post-thumbnail';
 	}
-
-	// Post requires password.
-	$post_password_required = post_password_required( $post->ID );
-	if ( $post_password_required ) {
-		$classes[] = 'post-password-required';
-	} elseif ( ! empty( $post->post_password ) ) {
-		$classes[] = 'post-password-protected';
+	if ( $product->get_post_password() ) {
+		$classes[] = post_password_required( $product->get_id() ) ? 'post-password-required' : 'post-password-protected';
 	}
-
-	// Post thumbnails.
-	if ( current_theme_supports( 'post-thumbnails' ) && $product->get_image_id() && ! is_attachment( $post ) && ! $post_password_required ) {
-		$classes[] = 'has-post-thumbnail';
+	if ( $product->is_on_sale() ) {
+		$classes[] = 'sale';
 	}
-
-	// Sticky for Sticky Posts.
-	if ( is_sticky( $post->ID ) ) {
-		if ( is_home() && ! is_paged() ) {
-			$classes[] = 'sticky';
-		} elseif ( is_admin() ) {
-			$classes[] = 'status-sticky';
-		}
+	if ( $product->is_featured() ) {
+		$classes[] = 'featured';
+	}
+	if ( $product->is_downloadable() ) {
+		$classes[] = 'downloadable';
+	}
+	if ( $product->is_virtual() ) {
+		$classes[] = 'virtual';
+	}
+	if ( $product->is_sold_individually() ) {
+		$classes[] = 'sold-individually';
+	}
+	if ( $product->is_taxable() ) {
+		$classes[] = 'taxable';
+	}
+	if ( $product->is_shipping_taxable() ) {
+		$classes[] = 'shipping-taxable';
+	}
+	if ( $product->is_purchasable() ) {
+		$classes[] = 'purchasable';
+	}
+	if ( $product->get_type() ) {
+		$classes[] = 'product-type-' . $product->get_type();
+	}
+	if ( $product->is_type( 'variable' ) && $product->get_default_attributes() ) {
+		$classes[] = 'has-default-attributes';
 	}
 
-	// Hentry for hAtom compliance.
-	$classes[] = 'hentry';
```

Comment:
I assume this is not needed any more, as the spec seems to be from ~2011.

Reply:
Nope we removed it anyway in `wc_product_post_class`